src/python/pants/backend/docker/util_rules/docker_build_context.py

```diff
@@ -66,6 +71,34 @@ def merge(self, other: Mapping[str, Mapping[str, str]]) -> DockerVersionContext:
         return DockerVersionContext.from_dict({**self, **other})
 
 
+class DockerContextFilesAcceptableInputsField(ABC, SourcesField):
+    """This is a meta field for the context files generator, to tell the codegen machinery what
+    source fields are good to use."""
+
+
+DockerContextFilesAcceptableInputsField.register(FileSourceField)
+DockerContextFilesAcceptableInputsField.register(ShellSourceField)
```

Comment:
Interestingggggggg! I would never have thought to use https://docs.python.org/3/library/abc.html#abc.ABCMeta.register

This is pretty advanced Python. I recommend explicitly linking to those docs in the docstring for `DockerContextFilesAcceptableInputsField`